tensorflow/lite/micro/kernels/select_test.cc

```diff
@@ -0,0 +1,257 @@
+
+  TfLiteSelectParams builtin_data;
+  const TfLiteRegistration registration = tflite::Register_SELECT_V2();
+  micro::KernelRunner runner(registration, tensors, tensors_size, inputs_array,
+                             outputs_array,
+                             reinterpret_cast<void*>(&builtin_data));
+
+  TF_LITE_MICRO_EXPECT_EQ(kTfLiteOk, runner.InitAndPrepare());
+  TF_LITE_MICRO_EXPECT_EQ(kTfLiteOk, runner.Invoke());
+}
+
+template <typename T>
+void ExpectEqual(int* dims, const T* expected_data, const T* output_data) {
+  TfLiteIntArray* dims_array = IntArrayFromInts(dims);
+  const int element_count = ElementCount(*dims_array);
+  for (int i = 0; i < element_count; ++i) {
+    TF_LITE_MICRO_EXPECT_EQ(expected_data[i], output_data[i]);
+  }
+}
+
+template <typename T>
+void ExpectNear(int* dims, const T* expected_data, const T* output_data) {
+  TfLiteIntArray* dims_array = IntArrayFromInts(dims);
+  const int element_count = ElementCount(*dims_array);
+  for (int i = 0; i < element_count; ++i) {
+    TF_LITE_MICRO_EXPECT_NEAR(expected_data[i], output_data[i], 1e-5);
+  }
+}
+
+}  // namespace testing
+}  // namespace tflite
+
+TF_LITE_MICRO_TESTS_BEGIN
+
+TF_LITE_MICRO_TEST(SelectBool) {
+  int inout_shape[] = {4, 1, 1, 1, 4};
+
+  const bool input1_data[] = {true, false, true, false};
+  const bool input2_data[] = {false, false, false, false};
+  const bool input3_data[] = {true, true, true, true};
+  const bool expected_output[] = {false, true, false, true};
+
+  bool output_data[4];
+  tflite::testing::TestSelect(inout_shape, input1_data, inout_shape,
+                              input2_data, inout_shape, input3_data,
+                              inout_shape, output_data);
+  tflite::testing::ExpectEqual(inout_shape, expected_output, output_data);
+}
+
+TF_LITE_MICRO_TEST(SelectFloat) {
+  int inout_shape[] = {4, 1, 1, 1, 4};
+
+  const bool input1_data[] = {true, false, true, false};
+  const float input2_data[] = {0.1, 0.2, 0.3, 0.4};
+  const float input3_data[] = {0.5, 0.6, 0.7, 0.8};
+  const float expected_output[] = {0.1, 0.6, 0.3, 0.8};
+
+  float output_data[4];
+  tflite::testing::TestSelect(inout_shape, input1_data, inout_shape,
+                              input2_data, inout_shape, input3_data,
+                              inout_shape, output_data);
+  tflite::testing::ExpectNear(inout_shape, expected_output, output_data);
+}
+
+TF_LITE_MICRO_TEST(SelectUInt8) {
+  int inout_shape[] = {4, 1, 1, 1, 4};
+
+  const bool input1_data[] = {false, true, false, false};
+  const uint8_t input2_data[] = {1, 2, 3, 4};
+  const uint8_t input3_data[] = {5, 6, 7, 8};
+  const uint8_t expected_output[] = {5, 2, 7, 8};
+
+  uint8_t output_data[4];
+  tflite::testing::TestSelect(inout_shape, input1_data, inout_shape,
+                              input2_data, inout_shape, input3_data,
+                              inout_shape, output_data);
+  tflite::testing::ExpectEqual(inout_shape, expected_output, output_data);
+}
+
```

Comment:
let's remove test cases for everything other than int8, int16, flaot32 (and any other type that you might need right now).